### ISSUE TITLE
*Ignore this post

### ISSUE THREAD

**Issue by K20qhuy** [fundamentalvision/Deformable-DETR#237]
With Cuda=10.1, pytorch=1.5.1 torchvision=0.6.1
I get this error below after sh make.sh
The TLDR: building 'MultiScaleDeformableAttention' extension UNSUCESSFUL

More DETAILED - key errors I interpreted:
#1
/usr/include/c++/7/bits/basic_string.h:6693:95:   required from here
/usr/include/c++/7/bits/basic_string.tcc:1067:16: error: cannot call member function ‘void std::basic_string<_CharT, _Traits, _Alloc>::_Rep::_M_set_sharable() [with _CharT = char32_t; _Traits = std::char_traits<char32_t>; _Alloc = std::allocator<char32_t>]’ without object
ninja: build stopped: subcommand failed.
Traceback (most recent call last):
...........................

#2
/home/bmvt/anaconda3/envs/deformable_detr_233/lib/python3.8/site-packages/torch/include/ATen/core/TensorBody.h:341:1: note: declared here
   T * data() const {
 ^ ~~
/usr/include/c++/7/bits/basic_string.tcc: In instantiation of ‘static std::basic_string<_CharT, _Traits, _Alloc>::_Rep* std::basic_string<_CharT, _Traits, _Alloc>::_Rep::_S_create(std::basic_string<_CharT, _Traits, _Alloc>::size_type, std::basic_string<_CharT, _Traits, _Alloc>::size_type, const _Alloc&) [with _CharT = char16_t; _Traits = std::char_traits<char16_t>; _Alloc = std::allocator<char16_t>; std::basic_string<.......................
.....................
/usr/include/c++/7/bits/basic_string.h:5063:24:   required from ‘static _CharT* std::basic_string<_CharT, _Traits, _Alloc>::_S_construct(_InIterator, _InIterator, const _Alloc&) [with _InIterator = const char16_t*; _CharT = char16_t; _Traits = std::char_traits<char16_t>; _Alloc = std::allocator<char16_t>]’
/usr/include/c++/7/bits/basic_string.tcc:656:134:   required from ‘std::basic_string<_CharT, _Traits, _Alloc>::basic_string(const _CharT*, std::basic_string<_CharT, _Traits, _Alloc>::size_type, const _Alloc&) [with _CharT = char16_t; _Traits = std::char_traits<char16_t>; _Alloc = std::allocator<char16_t>; std::basic_string<_CharT, _Traits, _Alloc>::size_type = long unsigned int]’
/usr/include/c++/7/bits/basic_string.h:6688:95:   required from here
/usr/include/c++/7/bits/basic_string.tcc:1067:16: error: cannot call member function ‘void std::basic_string<_CharT, _Traits, _Alloc>::_Rep::_M_set_sharable() [with _CharT = char16_t; _Traits = std::char_traits<char16_t>; _Alloc = std::allocator<char16_t>]’ without object
       __p->_M_set_sharable();
       ~~~~~~~~~^~
/usr/include/c++/7/bits/basic_string.tcc: In i........................

#3 ???????
...
/home/bmvt/Desktop/Huy_SVthayChau/DETopK_233/models/ops/src/cuda/ms_deform_im2col_cuda.cuh(261): warning: variable "q_col" was declared but never referenced
          detected during instantiation of "void ms_deformable_im2col_cuda(cudaStream_t, const scalar_t *, const int64_t *, const int64_t *, const scalar_t *, const scalar_t *, int, int, int, int, int, int, int, scalar_t *) [with scalar_t=double]" 
/home/bmvt/Desktop/Huy_SVthayChau/DETopK_233/models/ops/src/cuda/ms_deform_attn_cuda.cu(64): here

/home/bmvt/Desktop/Huy_SVthayChau/DETopK_233/models/ops/src/cuda/ms_deform_im2col_cuda.cuh(762): warning: variable "q_col" was declared but never referenced
          detected during instantiation of "void ms_deformable_col2im_cuda(cudaStream_t, const scalar_t *, const scalar_t *, const int64_t *, const int64_t *, const scalar_t *, const scalar_t *, int, int, int, int, int, int, int, scalar_t *, scalar_t *, scalar_t *) [with scalar_t=double]" 
/home/bmvt/Desktop/Huy_SVthayChau/DETopK_233/models/ops/src/cuda/ms_deform_attn_cuda.cu(134): here

/home/bmvt/Desktop/Huy_SVthayChau/DETopK_233/models/ops/src/cuda/ms_deform_im2col_cuda.cuh(872): warning: variable "q_col" was declared but never referenced
          detected during instantiation of "void ms_deformable_col2im_cuda(cudaStream_t, const scalar_t *, const scalar_t *, const int64_t *, const int64_t *, const scalar_t *, const scalar_t *, int, int, int, int, int, int, int, scalar_t *, scalar_t *, scalar_t *) [with scalar_t=double]" 
/home/bmvt/Desktop/Huy_SVthayChau/DETopK_233/models/ops/src/cuda/ms_deform_attn_cuda.cu(134): here

/home/bmvt/Desktop/Huy_SVthayChau/DETopK_233/models/ops/src/cuda/ms_deform_im2col_cuda.cuh(331): warning: variable "q_col" was declared but never referenced
          detected during instantiation of "void ms_deformable_col2im_cuda(cudaStream_t, const scalar_t *, const scalar_t *, const int64_t *, const int64_t *, const scalar_t *, const scalar_t *, int, int, int, int, int, int, int, scalar_t *, scalar_t *, scalar_t *) [with scalar_t=double]" 
/home/bmvt/Desktop/Huy_SVthayChau/DETopK_233/models/ops/src/cuda/ms_deform_attn_cuda.cu(134): here

/home/bmvt/Desktop/Huy_SVthayChau/DETopK_233/models/ops/src/cuda/ms_deform_im2col_cuda.cuh(436): warning: variable "q_col" was declared but never referenced
          detected during instantiation of "void ms_deformable_col2im_cuda(cudaStream_t, const scalar_t *, const scalar_t *, const int64_t *, const int64_t *, const scalar_t *, const scalar_t *, int, int, int, int, int, int, int, scalar_t *, scalar_t *, scalar_t *) [with scalar_t=double]" 
/home/bmvt/Desktop/Huy_SVthayChau/DETopK_233/models/ops/src/cuda/ms_deform_attn_cuda.cu(134): here

/home/bmvt/Desktop/Huy_SVthayChau/DETopK_233/models/ops/src/cuda/ms_deform_im2col_cuda.cuh(544): warning: variable "q_col" was declared but never referenced
          detected during instantiation of "void ms_deformable_col2im_cuda(cudaStream_t, const scalar_t *, const scalar_t *, const int64_t *, const int64_t *, const scalar_t *, const scalar_t *, int, int, int, int, int, int, int, scalar_t *, scalar_t *, scalar_t *) [with scalar_t=double]" 
/home/bmvt/Desktop/Huy_SVthayChau/DETopK_233/models/ops/src/cuda/ms_deform_attn_cuda.cu(134): here

/home/bmvt/Desktop/Huy_SVthayChau/DETopK_233/models/ops/src/cuda/ms_deform_im2col_cuda.cuh(649): warning: variable "q_col" was declared but never referenced
          detected during instantiation of "void ms_deformable_col2im_cuda(cudaStream_t, const scalar_t *, const scalar_t *, const int64_t *, const int64_t *, const scalar_t *, const scalar_t *, int, int, int, int, int, int, int, scalar_t *, scalar_t *, scalar_t *) [with scalar_t=double]" 
/home/bmvt/Desktop/Huy_SVthayChau/DETopK_233/models/ops/src/cuda/ms_deform_attn_cuda.cu(134): here

/home/bmvt/Desktop/Huy_SVthayChau/DETopK_233/models/ops/src/cuda/ms_deform_attn_cuda.cu: In function ‘at::Tensor ms_deform_attn_cuda_forwar............

#4 OTHERS:

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "setup.py", line 62, in <module>
    setup(
  File "/home/bmvt/anaconda3/envs/deformable_detr_233/lib/python3.8/site-packages/setuptools/__init__.py", line 108, in setup
    return distutils.core.setup(**attrs)
  File "/home/bmvt/anaconda3/envs/deformable_detr_233/lib/python3.8/site-packages/setuptools/_distutils/core.py", line 184, in setup
    return run_commands(dist)
  File "/home/bmvt/anaconda3/envs/deformable_detr_233/lib/python3.8/site-packages/setuptools/_distutils/core.py", line 200, in run_commands
    dist.run_commands()
  File "/home/bmvt/anaconda3/envs/deformable_detr_233/lib/python3.8/site-packages/setuptools/_distutils/dist.py", line 970, in run_commands
    self.run_command(cmd)
  File "/home/bmvt/anaconda3/envs/deformable_detr_233/lib/python3.8/site-packages/setuptools/dist.py", line 956, in run_command
    super().run_command(command)
  File "/home/bmvt/anaconda3/envs/deformable_detr_233/lib/python3.8/site-packages/setuptools/_distutils/dist.py", line 989, in run_command
    cmd_obj.run()
  File "/home/bmvt/anaconda3/envs/deformable_detr_233/lib/python3.8/site-packages/setuptools/_distutils/command/build.py", line 135, in run
    self.run_command(cmd_name)
  File "/home/bmvt/anaconda3/envs/deformable_detr_233/lib/python3.8/site-packages/setuptools/_distutils/cmd.py", line 316, in run_command
    self.distribution.run_command(command)
  File "/home/bmvt/anaconda3/envs/deformable_detr_233/lib/python3.8/site-packages/setuptools/dist.py", line 956, in run_command
    super().run_command(command)
  File "/home/bmvt/anaconda3/envs/deformable_detr_233/lib/python3.8/site-packages/setuptools/_distutils/dist.py", line 989, in run_command
    cmd_obj.run()
  File "/home/bmvt/anaconda3/envs/deformable_detr_233/lib/python3.8/site-packages/setuptools/command/build_ext.py", line 93, in run
    _build_ext.run(self)
  File "/home/bmvt/.local/lib/python3.8/site-packages/Cython/Distutils/old_build_ext.py", line 186, in run
    _build_ext.build_ext.run(self)
  File "/home/bmvt/anaconda3/envs/deformable_detr_233/lib/python3.8/site-packages/setuptools/_distutils/command/build_ext.py", line 359, in run
    self.build_extensions()
  File "/home/bmvt/anaconda3/envs/deformable_detr_233/lib/python3.8/site-packages/torch/utils/cpp_extension.py", line 603, in build_extensions
    build_ext.build_extensions(self)
  File "/home/bmvt/.local/lib/python3.8/site-packages/Cython/Distutils/old_build_ext.py", line 195, in build_extensions
    _build_ext.build_ext.build_extensions(self)
  File "/home/bmvt/anaconda3/envs/deformable_detr_233/lib/python3.8/site-packages/setuptools/_distutils/command/build_ext.py", line 479, in build_extensions
    self._build_extensions_serial()
  File "/home/bmvt/anaconda3/envs/deformable_detr_233/lib/python3.8/site-packages/setuptools/_distutils/command/build_ext.py", line 505, in _build_extensions_serial
    self.build_extension(ext)
  File "/home/bmvt/anaconda3/envs/deformable_detr_233/lib/python3.8/site-packages/setuptools/command/build_ext.py", line 254, in build_extension
    _build_ext.build_extension(self, ext)
  File "/home/bmvt/anaconda3/envs/deformable_detr_233/lib/python3.8/site-packages/setuptools/_distutils/command/build_ext.py", line 560, in build_extension
    objects = self.compiler.compile(
  File "/home/bmvt/anaconda3/envs/deformable_detr_233/lib/python3.8/site-packages/torch/utils/cpp_extension.py", line 428, in unix_wrap_ninja_compile
    _write_ninja_file_and_compile_objects(
  File "/home/bmvt/anaconda3/envs/deformable_detr_233/lib/python3.8/site-packages/torch/utils/cpp_extension.py", line 1158, in _write_ninja_file_and_compile_objects
    _run_ninja_build(
  File "/home/bmvt/anaconda3/envs/deformable_detr_233/lib/python3.8/site-packages/torch/utils/cpp_extension.py", line 1436, in _run_ninja_build
    raise RuntimeError(message)
RuntimeError: Error compiling objects for extension

**Comment by K20qhuy** [comment]
me as the OP, already ignored this post

**Comment by K20qhuy** [comment]
me as the OP, already ignored this post
